hadoop-ozone/common/src/main/java/org/apache/hadoop/ozone/om/protocolPB/OzoneManagerProtocolClientSideTranslatorPB.java

```diff
@@ -1354,7 +1354,18 @@ public ListSnapshotResponse listSnapshot(
         .map(snapshotInfo -> SnapshotInfo.getFromProtobuf(snapshotInfo))
         .collect(Collectors.toList());
 
-    return new ListSnapshotResponse(snapshotInfos, response.hasLastSnapshot() ? response.getLastSnapshot() : null);
+    String lastSnapshot = null;
+
+    if (response.hasLastSnapshot()) {
+      lastSnapshot = response.getLastSnapshot();
+    } else if (snapshotInfos.size() == maxListResult) {
```

Comment:
We are still not aware if there are more entries(total number of snapshots % maxListResult == 0). We are still back to the same problem where we would be hitting another OM request after the last page. I am not sure how we can avoid this.

Reply:
I am unsure how to avoid that without breaking any forward and backward compatibility.

Reply:
@sumitagrawl Yes, it makes sense. It should be similar to what we have in proto basically.

Reply:
Had an offline discussion. Since the change in proto is just adding a new property, it is backward and forward-compatible.